geps/gep-713/index.md

```diff
@@ -1,1157 +1,747 @@
 
-**Ana**: _Nope. Definitely didn’t do that._
+### Declared targets versus Effective targets
 
-She types `kubectl get retrypolicy -n baker` and gets a permission error.
+A kind specified in the target reference of a meta resource can be the actual kind of object whose behavior the meta resource intends to augment or an indirection to targeting other kinds the object is hierarchically related to.
 
-**Ana**: _Huh, I actually don’t have permissions for RetryPolicy._ 🤔
+E.g. targeting a Gateway API `Gateway` object with a meta resource can be a way to augment the behavior of the `Gateway` object itself (e.g. reconcile cloud infrastructure provider settings from the spec declared by the `Gateway` according the rules specified by the meta resource attached to the `Gateway`) or a means to augment the behavior of all `HTTPRoute` objects attached to the `Gateway` (in a way that every new `HTTPRoute` that gets created or modified so it enters the context of the `Gateway` is automatically put in the scope of the meta resource.)
 
-**Chihiro**: 🤷 _Feels like you should but OK, guess that can’t be it._
+The target kinds specified in the target references of a meta resource are referred to as *Declared target* kinds. When declared targets are not equal to the actual targets augmented by the meta resource, but rather they serve as a means for reaching other levels (typically lower level) of related object kinds, these other kinds of target objects whose behavior are actually augmented by the meta resource are referred to as *Effective target* kinds.
 
-Minutes pass while both look at logs.
+### Hierarchy of target kinds and mechanics for calculating Effective meta resources
 
-**Chihiro**: _I’m an idiot. There’s a RetryPolicy for the whole namespace –
-sorry, too many policies in the dashboard and I missed it. Deleting that since
-you don’t want retries._
+Target kinds MUST be organized in a well-known hierarchy of target kinds, from the least specific declared target kinds to the most specific effective ones. This hierarchy shall thus induce a logical representation of the relationships between instances of target objects, in the form of a Directed Acyclic Graph (DAG) whose roots are the least specific objects and the leaves are the most specific ones (and ultimately the effective targets of the meta resources).
```

Comment:
```suggestion
Target kinds MUST be organized in a well-known hierarchy of target kinds, from the least specific target kinds to the most specific ones, across both declared and effective target kinds.

The best way to visualize this hierarchy is in the form of a Directed Acyclic Graph (DAG) whose roots are the least specific objects and the leaves are the most specific ones (and ultimately the effective targets of the meta resources).
```
Simplifying the text a bit again.

Reply:
We must be careful not to conflate two concepts here: the hierarchy of target kinds and the DAG of instances (which is induced by the hierarchy.)

The hierarchy you define once for your domain, whereas the DAG maps a given state in a given moment in time.

Reply:
Tried something in between original and suggestion. I hope that works.

Reply:
That's better, thanks.